converter/src/main/java/gov/cms/qpp/conversion/model/error/Detail.java

```diff
@@ -31,42 +39,51 @@ public Detail() {
 	 * Copy constructor
 	 */
 	public Detail(Detail detail) {
-		this(detail.getMessage(), detail.getPath(), detail.getValue(), detail.getType());
+		errorCode = detail.errorCode;
+		message = detail.message;
+		path = detail.path;
+		value = detail.value;
+		type = detail.type;
 	}
 
 	/**
-	 * Constructs a {@code Detail} with just a description.
-	 *
-	 * @param text A description of the error.
+	 * Creates a mutable Detail based on the given error and node
 	 */
-	public Detail(String text) {
-		this.message = text;
+	public static Detail forErrorAndNode(LocalizedError error, Node node) {
+		Objects.requireNonNull(node, "node");
+
+		Detail detail = forErrorCode(error);
+		detail.setPath(node.getPath());
+		return detail;
 	}
 
 	/**
-	 * Constructs a {@code Detail} with a description and an path to point where the error is in the original document.
-	 *
-	 * @param text A description of the error.
-	 * @param path A path to where the error is.
+	 * Creates a mutable Detail based on the given error
+	 * @param code
+	 * @return
 	 */
-	public Detail(String text, String path) {
-		this.message = text;
-		this.path = path;
+	public static Detail forErrorCode(LocalizedError error) {
+		Objects.requireNonNull(error, "error");
+
+		Detail detail = new Detail();
+		detail.setErrorCode(error.getErrorCode().ordinal());
```

Comment:
I am not a fan of using the ordinal value.  :\  I outlined my reasoning in a previous comment.  https://github.com/CMSgov/qpp-conversion-tool/pull/417#discussion_r148096056

Reply:
Fixed